template/src/lib/service.broker.ts

```diff
@@ -0,0 +1,88 @@
+import { TypedServiceBroker } from 'moleculer-service-ts';
+import { Service, Middleware, ServiceBroker, EventSchema } from 'moleculer';
+import {
+  MikroConnector,
+  DatabaseContextManager,
+  MoleculerMikroContext
+} from 'moleculer-context-db';
+import { ServiceAction, ServiceEvent, ServiceName } from '../service.types';
+import { entities } from '../entities';
+import { config } from './env';
+
+import { brokerConfig } from './moleculer/broker.config';
+import { {{capitalizedServiceName}}Service } from '../{{serviceName}}.service';
+
+let service: Service;
+let started = false;
+
+export const broker: TypedServiceBroker<
+  ServiceAction,
+  ServiceEvent,
+  ServiceName
+> = new TypedServiceBroker(brokerConfig);
+
+export const dbConnector: MikroConnector = new MikroConnector();
+
+const LogMiddleware: Middleware = {
+  localAction(next: any) {
+    return function logAction(this: ServiceBroker, ctx: MoleculerMikroContext) {
+      broker.logger.info(
+        `Action '${ctx.action?.name}'.  '${ctx.caller}'. CallerNode '${ctx.nodeID}'.`
+      );
+      return next(ctx);
+    };
+  },
+  localEvent(next: any, event: EventSchema) {
+    return function logEvent(payload: any, sender: string, eventName: string) {
+      broker.logger.info(`Event ${event.name}. From service '${sender}'`);
+      return next(payload, sender, eventName);
+    };
+  }
+};
+
+export async function startService(): Promise<Service> {
+  if (started) {
+    throw new Error(
+      `startService() already called. Now you can only call stopService().`
+    );
+  }
+  started = true;
+
+  broker.middlewares.add(LogMiddleware);
+
+  await dbConnector.init({
```

Comment:
Not all services will need a db connector, e.g. API gateway. So need for Mikro-ORM there at all. How will it work there? DB connector should be made optional.

Reply:
I will just put db connector outside lib